uikit/viewmodels/src/main/java/com/sygic/maps/uikit/viewmodels/navigation/signpost/BaseSignpostViewModel.kt

```diff
@@ -0,0 +1,84 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sygic.maps.uikit.viewmodels.navigation.signpost
+
+import androidx.annotation.CallSuper
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModel
+import com.sygic.maps.uikit.viewmodels.R
+import com.sygic.maps.uikit.viewmodels.common.extensions.getDirectionDrawable
+import com.sygic.maps.uikit.viewmodels.common.extensions.getDistanceWithUnits
+import com.sygic.maps.uikit.viewmodels.common.regional.RegionalManager
+import com.sygic.maps.uikit.viewmodels.common.regional.units.DistanceUnit
+import com.sygic.maps.uikit.viewmodels.navigation.signpost.direction.DirectionManeuverType
+import com.sygic.maps.uikit.views.common.utils.TextHolder
+import com.sygic.sdk.navigation.NavigationManager
+import com.sygic.sdk.navigation.warnings.DirectionInfo
+
+@Suppress("unused", "MemberVisibilityCanBePrivate")
+abstract class BaseSignpostViewModel(
+    private val regionalManager: RegionalManager,
+    private val navigationManager: NavigationManager
+) : ViewModel(), NavigationManager.OnDirectionListener {
+
+    val distance: MutableLiveData<String> = MutableLiveData()
+    val primaryDirection: MutableLiveData<Int> = MutableLiveData()
+    val secondaryDirection: MutableLiveData<Int> = MutableLiveData()
+    val secondaryDirectionText: Int = R.string.then
+    val secondaryDirectionContainerVisible: MutableLiveData<Boolean> = MutableLiveData(false)
+    val instructionText: MutableLiveData<TextHolder> = MutableLiveData(TextHolder.empty())
+
+    protected var distanceUnit: DistanceUnit = DistanceUnit.KILOMETERS
+    protected val directionInfo: MutableLiveData<DirectionInfo?> = MutableLiveData()
+
+    private val distanceUnitObserver = Observer<DistanceUnit> { distanceUnit = it }
+    private val directionInfoObserver = Observer<DirectionInfo?> { directionInfo ->
+        directionInfo?.let {
+            distance.value = it.getDistanceWithUnits(distanceUnit)
+            primaryDirection.value = it.getDirectionDrawable(DirectionManeuverType.PRIMARY)
+            secondaryDirection.value = it.getDirectionDrawable(DirectionManeuverType.SECONDARY)
+            secondaryDirectionContainerVisible.value = it.secondary.isValid
+        }
+    }
+
+    init {
+        navigationManager.addOnDirectionListener(this)
+        directionInfo.observeForever(directionInfoObserver)
+        regionalManager.distanceUnit.observeForever(distanceUnitObserver)
+    }
+
+    @CallSuper
+    override fun onDirectionInfoChanged(directionInfo: DirectionInfo) {
+        this.directionInfo.value = directionInfo
```

Comment:
toto protected `directionInfo` sluzi aj na ine ako len na observovanie `directionInfoObserver`om? Lebo ten kod co je v observeri by mohol byt priamo tu v `onDirectionInfoChanged` ak nie

Reply:
to jsem už jen opomněl z původních experimentů, odstaním, díky.